### PR TITLE
release: v1.14.0

### PR DESCRIPTION
**Pull request by abernix**
**This particular PR should be true-merged to `main`.**

This PR represents the merge to `main` of the v1.14.0 release and follows up the [release prep](https://github.com/apollographql/router/pull/2911) PR.

This PR is **mainly a merge commit**, so reviewing every individual commit shown below is not necessary since those have been reviewed in their own PR.

**However!** Some things to review on this PR:

- Does this PR target the right branch? (usually, `main`)
- Are the appropriate **version bumps** and **release note edits** in the end of the commit list (or within the last few commits).  In other words, did the release prep PR actually land on this branch?

If those things look good, this PR is good to merge.

### CODE REVIEW

**Comment by abernix** [comment]
Note: This PR should be marked as "Ready for review" only AFTER #2911 has merged into it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,68 @@ All notable changes to Router will be documented in this file.
...
+
+By [@o0ignition0o](https://github.com/o0ignition0o) in https://github.com/apollographql/router/pull/2863
+
+## 📃 Configuration
+
+### Helm: Router chart now supports `extraLabels` for Deployments/Pods
+
+Our Helm chart now supports a new value called `extraLabels`, which enables chart users to add custom labels to the Router Deployment and its Pods.
+
+By [@gscheibel(https://github.com/gscheibel/) in https://github.com/apollographql/router/pull/2903
```
Comment (o0Ignition0o) [inline CHANGELOG.md:41]:
```suggestion
By [@gscheibel](https://github.com/gscheibel/) in https://github.com/apollographql/router/pull/2903
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,68 @@ All notable changes to Router will be documented in this file.
...
+
+### Helm: Router chart now supports `extraContainers` to run sidecars
+
+Our Helm chart now supports `extraContainers` in an effort to simplify the ability to run containers alongside Router containers (sidecars) which is a useful pattern for [coprocessors](https://www.apollographql.com/docs/router/customizations/coprocessor/).
+
+By [@pcarrier(https://github.com/pcarrier) in https://github.com/apollographql/router/pull/2881
```
Comment (o0Ignition0o) [inline CHANGELOG.md:47]:
```suggestion
By [@pcarrier](https://github.com/pcarrier) in https://github.com/apollographql/router/pull/2881
```

**Comment by abernix** [comment]
Thanks! Auto merge enabled.

**Review by o0Ignition0o — APPROVED** [review]

**Review by bnjjj — APPROVED** [review]

**Pull request merged**